pkg/ingester/limiting.go

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ingester
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/prometheus/procfs"
+	"github.com/weaveworks/common/httpgrpc"
+)
+
+// This is the closest fitting Prometheus API error code for requests rejected due to limiting.
+const queryLimitingCode = http.StatusServiceUnavailable
+
+func scanCgroupV2() (float64, uint64, error) {
```

Comment:
Isn't there any small library we can use to read these stats from cgroups, instead of building it ourselves?

Reply:
Not sure, I could look.

Reply:
Failed to find any simple library for it, and do we really gain anything by using a library instead of the current, dead simple implementation (which is also based on cgroup v1/v2 references)?